koschei/locks.py

```diff
@@ -97,4 +98,11 @@ def pg_session_lock(db, namespace, key, block=True, shared=False):
         yield
     finally:
         if locked:
-            pg_unlock(db, namespace, key, shared=shared)
+            try:
+                pg_unlock(db, namespace, key, shared=shared)
+            # If the transaction is aborted, the unlock fails. We need to
+            # start a new trasaction by rolling back.
+            except InternalError:
+                db.rollback()
+                pg_unlock(db, namespace, key, shared=shared)
```

Comment:
What if this unlock also fails? Why not simply close the session in this case and let the server unlock all locks by itself?

Reply:
Closing the db.session, won't close the connection (postgres session). In order to close the connection, it would need to cooperate with `KoscheiBackendSession`. So I'd need to couple those two somehow:
- Make the lock functions methods of the koschei session
- Pass the koschei session as argument instead of the DB session